README.md

```diff
@@ -1 +1,174 @@
+
+Here is the list of the core modules for v5.1:
+
+* [`assert`](https://nodejs.org/api/assert.html): A simple set of assertion test
+* [`cluster`](https://nodejs.org/api/cluster.html): Interface to launch a cluster (multiple) of Node processes to increase performance 
+* [`crypto`](https://nodejs.org/api/crypto.html): Cryptographic functionality for OpenSSL, HMAC, etc.
+* [`events`](https://nodejs.org/api/events.html): Interface for asynchronous event-driven architecture (used by many other core modules)
+* [`fs`](https://nodejs.org/api/fs.html): File system I/O methods
+* [`http`](https://nodejs.org/api/http.html): Interface for HTTP servers and clients
+* [`net`](https://nodejs.org/api/net.html): Interface for networking which can be used to implement any protocol, not just HTTP (`http` uses `net`)
+* [`os`](https://nodejs.org/api/os.html): Basic operating-system utilities
+* [`path`](https://nodejs.org/api/path.html): Utilities for handling and transmitting file paths on various platforms.
+* [`querystring`](https://nodejs.org/api/querystring.html): Utilities for working with query strings (e.g., `?key1=value1&key2=value2` in a URL)
+* [`stream`](https://nodejs.org/api/stream.html): Abstract interface implemented by various objects.
+* [`url`](https://nodejs.org/api/url.html): Utilities for URL resolution and parsing.
+* [`util`](https://nodejs.org/api/util.html): Utilities which are mostly used by other core Node modules
+
+Are your head is about to explode from this list of core modules? Stop the panic. You don't need to remember all of them. Just know that they are out there when you have a task at hand which requires you to require that module (pun intended), e.g., `crypto` for hashing of the passwords.
+
+In the next section, we'll show you some examples of `fs`, `path`, and `url`.
+
+## fs
+
+You can read a file asynchronously with `fs.readFile` and write to a file with `fs.writeFile`, e.g.,
+
+```js
+var callback = function(error, data) {
+  console.log(data, error)
+}
+fs.readFile('data.json', 'utf8', callback)
+
+fs.writeFile('message.txt', 'Hello Node.js', function (error) {
+  console.log(error)
+})
+```
+
+## path
+
+We all want to develop application that work across platform with minimal or no code modifications, because then we gain a broader user base and flexibility. Node work out of the box with Unix, Linux, Mac OS X and Windows systems.
+
+But what would happen if I run a code with a Unix-like path on a Windows machine?
+
+```js
+fs.readFile('configurations/data/models/user.json', 'utf8', callback)
+```
+
+It probably won't do anything. So we need to have if/else conditions in our code:
+
+```js
+var platform = require('os').platform()
+
+if (platform == 'win32') {
+  var filePath = 'configurations\\data\\models\\'
+} else { // assume platform is darwin or similar
+  var filePath  = 'configurations/data/models/'
+}
+fs.readFile(filePath + 'user.json', 'utf8', callback)
+```
+
```

Comment:
walk students through this code. we're requiring the `OS` core module that deals with operating-system utilities, we call the platform function which returns a string (for windows "win32", for linxux "__", for OSX, "__")....